pkg/cfn/manager/api.go

```diff
@@ -178,8 +178,23 @@ func (c *StackCollection) CreateStack(ctx context.Context, stackName string, res
 
 // createClusterStack creates the cluster stack
 func (c *StackCollection) createClusterStack(ctx context.Context, stackName string, resourceSet builder.ResourceSetReader, errCh chan error) error {
+	// Do a preliminary check, so we don't end up waiting for nothing.
+	stack, err := c.DescribeStack(ctx, &types.Stack{
+		StackName: &stackName,
+	})
+	if err != nil && !IsStackDoesNotExistError(err) {
+		return err
+	}
+	if stack != nil {
+		switch stack.StackStatus {
+		case types.StackStatusCreateComplete, types.StackStatusDeleteComplete:
+			close(errCh) // no need to wait on this task anymore
+			return nil
+		}
```

Comment:
This will result in a bug. If a cluster with the same name already exists, the current behaviour is to throw an error saying that it already exists. But this will skip that check and attempt to create a Kubernetes client without having the values for the endpoint and CA data, failing with a cryptic error. 

(I've had this comment typed out for hours but forgot to hit submit).